docs/ase-file-specs.md

```diff
@@ -338,13 +340,51 @@ there is an User Data Chunk at the first frame after the Palette Chunk.
     DWORD       Flags
                   1 = Has text
                   2 = Has color
+                  4 = Has properties
     + If flags have bit 1
       STRING    Text
     + If flags have bit 2
       BYTE      Color Red (0-255)
       BYTE      Color Green (0-255)
       BYTE      Color Blue (0-255)
       BYTE      Color Alpha (0-255)
+    + If flags have bit 4
+      DWORD     Number of properties maps
+      + For each properties map:
+        DWORD     Properties maps key (Zero or extension's Entry ID (see External Files Chunk))
+        DWORD     Number of properties
+        + For each property:
+          STRING    Property name
+          WORD      Property type
+          + If property type is bool (0x0001)
+            BYTE    False (0x00) or true (any value different than 0x00)
+          + If property type is unsigned char (0x0002)
+            BYTE
+          + If property type is short (0x0003)
+            SHORT
+          + If property type is unsigned short (0x0004)
+            WORD
+          + If property type is int (0x0005)
+            LONG
+          + If property type is unsigned int (0x0006)
+            DWORD
+          + If property type is long long (0x0007)
+            LONG64
+          + If property type is unsigned long long (0x0008)
+            QWORD
+          + If property type is fixmath::fixed (0x0009)
+            FIXED
+          + If property type is std::string (0x000A)
+            STRING
+          + If property type is std::vector (0x0100)
+            DWORD     Number of elements
+            WORD      Element's type
+            BYTE[]    As many values as the number of elements indicates
+                      Structure depends on the element's type
+          + If property type is UserData::Properties (0x0101)
```

Comment:
Why this kind of data is needed? are not we in the properties map already?

Reply:
Didn't think about the possibility to save maps in one property. Probably the serialization could be simplified in such a way that there are two new types of properties (one of them a virtual property): 1) Open map, 2) Close map

So all the properties between Open Map <-> Close Map belong to the key specified in Open Map.

We have to talk with @Gasparoken about so the Lua API gives this possibility.